test/api/v3/integration/challenges/POST-challenges_challengeId_join.test.js

```diff
@@ -0,0 +1,113 @@
+import {
+  generateUser,
+  createAndPopulateGroup,
+  translate as t,
+} from '../../../../helpers/api-v3-integration.helper';
+import { v4 as generateUUID } from 'uuid';
+
+describe('POST /challenges/:challengeId/join', () => {
+  it('returns error when challengeId is not a valid UUID', async () => {
+    let user = await generateUser({ balance: 1});
+
+    await expect(user.post('/challenges/test/join')).to.eventually.be.rejected.and.eql({
+      code: 400,
+      error: 'BadRequest',
+      message: t('invalidReqParams'),
+    });
+  });
+
+  it('returns error when challengeId is not for a valid challenge', async () => {
+    let user = await generateUser({ balance: 1});
+
+    await expect(user.post(`/challenges/${generateUUID()}/join`)).to.eventually.be.rejected.and.eql({
+      code: 404,
+      error: 'NotFound',
+      message: t('challengeNotFound'),
+    });
+  });
+
+  context('Joining a valid challenge', () => {
+    let groupLeader;
+    let group;
+    let challenge;
+    let authorizedUser;
+
+    beforeEach(async () => {
+      let populatedGroup = await createAndPopulateGroup({
+        members: 1,
+      });
+
+      groupLeader = populatedGroup.groupLeader;
+      group = populatedGroup.group;
+      authorizedUser = populatedGroup.members[0];
+
+      challenge = await groupLeader.post('/challenges', {
+        groupId: group._id,
+        name: 'Test Challenge',
+        shortName: 'T.C.',
+      });
+    });
+
+    it('returns error when user doesn\'t have permissions to access the challenge', async () => {
+      let unauthorizedUser = await generateUser();
+
+      await expect(unauthorizedUser.post(`/challenges/${challenge._id}/join`)).to.eventually.be.rejected.and.eql({
+        code: 404,
+        error: 'NotFound',
+        message: t('challengeNotFound'),
+      });
+    });
+
+    it('adds challenge to user challenges', async () => {
+      await authorizedUser.post(`/challenges/${challenge._id}/join`);
+
+      await authorizedUser.sync();
+
+      expect(authorizedUser).to.have.property('challenges').to.include(challenge._id);
+    });
+
+    it('returns error when user has already joined the challenge', async () => {
+      await authorizedUser.post(`/challenges/${challenge._id}/join`);
+
+      await expect(authorizedUser.post(`/challenges/${challenge._id}/join`)).to.eventually.be.rejected.and.eql({
+        code: 401,
+        error: 'NotAuthorized',
+        message: t('userAlreadyInChallenge'),
+      });
+    });
+
+    it('increases memberCount of challenge', async () => {
+      let oldMemberCount = challenge.memberCount;
+
+      await authorizedUser.post(`/challenges/${challenge._id}/join`);
+
+      challenge = await authorizedUser.get(`/challenges/${challenge._id}`); // Switch to sync syntax when it's implemented for challenges
+
+      expect(challenge).to.have.property('memberCount', oldMemberCount + 1);
+    });
+
+    it('syncs challenge tasks to joining user', async () => {
```

Comment:
here I'd say to check that `task.challenge` is correct, so `task.challenge.id === challenge._id` and maybe that `task.challenge.taskId` correctly matches the original task challenge. But if we decide that `syncToUser` will be unit tested, then we can skip these tests. @crookedneighbor ?